mac/Keyman4MacIM/Keyman4MacIM/KMInputMethodAppDelegate.m

```diff
@@ -1165,42 +1178,111 @@ - (void)postKeyboardEventWithSource: (CGEventSourceRef)source code:(CGKeyCode) v
     }
 }
 
+
+// TODO: This seriously needs to be refactored out of the app delegate and into
+//       a keyboard install module
```

Comment:
Aside from a complete refactoring, would it be possible to have some of the new version-check specific code be isolated to a separate function so that this one doesn't become more unwieldy?